lib/g_searcher_web/controllers/search_result_controller.ex

```diff
@@ -3,13 +3,31 @@ defmodule GSearcherWeb.SearchResultController do
 
   alias GSearcher.{SearchResults, SearchResultURLs}
   alias GSearcherWeb.ErrorHandler
+  alias GSearcherWeb.Helpers.SearchHelper
+  alias GSearcherWeb.Validators.{ParamsValidator, SearchResultParams}
 
   def index(conn, params) do
     %{id: user_id} = conn.assigns.current_user
 
-    search_results = SearchResults.list_search_results_by_user_id(user_id, params["query"])
+    with {:ok, query_params} <- SearchHelper.parse_query(params["query"]),
+         {:ok, validated_params} <-
+           ParamsValidator.validate(query_params, as: SearchResultParams),
+         search_results <- SearchResults.list_search_results_by_user_id(user_id, validated_params) do
+      render(conn, "index.html", search_results: search_results)
+    else
+      {:error, :no_query} ->
```

Comment:
It is weird to put the path that is not error handling here. How about use pattern matching for `index` action?
```elixir
def index(conn, %{"query" => query}) when is_binary(query) do
  # Validate params and do a search
end

def index(conn, params) do
  %{id: user_id} = conn.assigns.current_user

  search_results = SearchResults.list_search_results_by_user_id(user_id)        
  
  render(conn, "index.html", search_results: search_results)
end
```

Reply:
That's a great idea! Thanks 😄 .
I've fixed it in 0349945a85fcf825a9e941873e03ea6059c4ce49